### PR TITLE
Use new design and add user flows as dummy pages WD-8469

### PR DESCRIPTION
**Pull request by edlerd**
* adjusted to new design
* added dummy pages for register, reset, device and setup

### CODE REVIEW

File: rockcraft.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: identity-platform-login-ui
 
 base: bare
-build-base: ubuntu:22.04
+build-base: ubuntu@22.04
```
Comment (shipperizer) [inline rockcraft.yaml:4]: can we separate this into another commit?
or simply drop it and we ll chuck it in the main branch and u can rebase

**Review by shipperizer — COMMENTED** [review]
it is quite a lot of code in just one commit, any chance to split it into smaller ones?

**Comment by edlerd** [comment]
> it is quite a lot of code in just one commit, any chance to split it into smaller ones?

It's not that many code changes, most of the lines are due to dependency updates, which are a prerequisite for other changes. So factoring them out into individual PRs is painful, because they would rely on each other.

File: ui/pages/login.tsx [hunk 2]
```diff
@@ -96,27 +96,14 @@ const Login: NextPage = () => {
 
           return Promise.reject(err);
         }),
-    [flow, router]
+    [flow, router],
   );
+  const reqDomain = flow?.return_to ? new URL(flow.return_to).hostname : "";
```
Comment (nsklikas) [inline ui/pages/login.tsx:101]: This is not correct. This `return_to` always points to Hydra, to get info about the client you would have to look into `flow.oauth2_login_request.client`. There are 2 fields there that we can use `client_name` which is supposed to be the human readable name of the client and `client_uri` which is the client's homepage (see https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata for more info)

Comment (edlerd) [reply ui/pages/login.tsx]: Good point, thanks for the clarification. I changed this to construct a string with client name and client url if present.

File: rockcraft.yaml (hunk 1 above)
Comment (edlerd) [reply rockcraft.yaml:4]: I factored this into a dedicated commit.

**Review by nsklikas — COMMENTED** [review]
LGTM other than the comment I made on the login page. One thing I don't understand is what the plan is for these registration pages, the fields are going to come from Kratos and I would expect that we are going to ask for more attributes rather than just username and password. We can iterate on this further when we do the work on the backend so I have no problem with merging this once the issue on the login page is fixed.

Here are some screen shots on what the new pages look like:
Login Page:
image

OIDC Error:
image

Register Secure:
image

Register Complete:
image

Register Email:
image

Register Password:
image

Register Secure:
image

Reset Check:
image

Reset Complete:
image

Reset Email:
image

Reset Password:
image

Setup Complete:
image

Setup Password:
image

Setup Secure:
image

**Comment by shipperizer** [comment]
@edlerd looks good to me afaict, what is the policy for frontend tests?how do we catch broken UI?

File: ui/components/NodeInputSubmit.tsx [hunk 3]
```diff
@@ -9,15 +9,59 @@ export const NodeInputSubmit: FC<NodeInputProps> = ({
...
+    }
+    if (value.toLowerCase().startsWith("google")) {
+      return "logos/Google.svg";
+    }
+    if (value.toLowerCase().startsWith("microsoft")) {
+      return "logos/Microsoft.svg";
+    }
+    if (value.toLowerCase().startsWith("okta")) {
+      return "logos/Okta.svg";
+    }
```
Comment (nsklikas) [inline ui/components/NodeInputSubmit.tsx:28]: There is no Okta provider type in Kratos, let's remove this and the Okta icon logo  (see https://www.ory.sh/docs/kratos/social-signin/overview)

File: ui/public/logos/Okta.svg
Comment (nsklikas) [inline ui/public/logos/Okta.svg:1]: There is not Okta provider type in Kratos, let's remove this (see https://www.ory.sh/docs/kratos/social-signin/overview)

**Review by nsklikas — COMMENTED** [review]
One more minor thing and I think we can merge it

File: ui/components/NodeInputSubmit.tsx (hunk 3 above)
Comment (edlerd) [reply ui/components/NodeInputSubmit.tsx]: removed the entry for okta.

File: ui/public/logos/Okta.svg
Comment (edlerd) [reply ui/public/logos/Okta.svg:1]: removed this asset

**Review by nsklikas — APPROVED** [review]
LGTM

**Review by shipperizer — APPROVED** [review]

**Comment by lukasSerelis** [comment]
Looks good. MFA options obviously missing but I assume this is out of scope at this stage
